shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/refresher/type/CreateTableStatementSchemaRefresherTest.java

```diff
@@ -31,90 +36,79 @@
+        when(materials.getDatabaseType()).thenReturn(new SQLServerDatabaseType());
         refresh(createTableStatement);
     }
     
+    // TODO add more tests for tables with table rule
     private void refresh(final CreateTableStatement createTableStatement) throws SQLException {
-        ShardingSphereSchema schema = ShardingSphereSchemaBuildUtil.buildSchema();
         createTableStatement.setTable(new SimpleTableSegment(new TableNameSegment(1, 3, new IdentifierValue("t_order_0"))));
+        Map<String, DataSource> dataSourceMap = mock(HashMap.class);
+        when(materials.getDataSourceMap()).thenReturn(dataSourceMap);
+        DataSource dataSource = mock(DataSource.class);
+        when(dataSourceMap.get(eq("ds"))).thenReturn(dataSource);
+        Connection connection = mock(Connection.class);
+        when(dataSource.getConnection()).thenReturn(connection);
+        DatabaseMetaData metaData = mock(DatabaseMetaData.class);
+        when(connection.getMetaData()).thenReturn(metaData);
+        ResultSet resultSet = mock(ResultSet.class);
+        when(metaData.getTables(any(), any(), any(), any())).thenReturn(resultSet);
+        when(resultSet.next()).thenReturn(false);
+        ShardingSphereSchema schema = ShardingSphereSchemaBuildUtil.buildSchema();
         SchemaRefresher<CreateTableStatement> schemaRefresher = new CreateTableStatementSchemaRefresher();
-        SchemaBuilderMaterials materials = mock(SchemaBuilderMaterials.class);
         schemaRefresher.refresh(schema, Collections.singleton("ds"), createTableStatement, materials);
         assertTrue(schema.containsTable("t_order_0"));
     }
-    
-    @Test
-    public void refreshWithUnConfiguredForMySQL() throws SQLException {
-        MySQLCreateTableStatement createTableStatement = new MySQLCreateTableStatement();
-        createTableStatement.setNotExisted(false);
-        refreshWithUnConfigured(createTableStatement);
-    }
-    
-    @Test
-    public void refreshWithUnConfiguredForOracle() throws SQLException {
-        OracleCreateTableStatement createTableStatement = new OracleCreateTableStatement();
-        refreshWithUnConfigured(createTableStatement);
-    }
-    
-    @Test
-    public void refreshWithUnConfiguredForPostgreSQL() throws SQLException {
-        PostgreSQLCreateTableStatement createTableStatement = new PostgreSQLCreateTableStatement();
-        createTableStatement.setNotExisted(false);
-        refreshWithUnConfigured(createTableStatement);
-    }
-    
-    @Test
-    public void refreshWithUnConfiguredForSQL92() throws SQLException {
-        SQL92CreateTableStatement createTableStatement = new SQL92CreateTableStatement();
-        refreshWithUnConfigured(createTableStatement);
-    }
-    
-    @Test
-    public void refreshWithUnConfiguredForSQLServer() throws SQLException {
-        SQLServerCreateTableStatement createTableStatement = new SQLServerCreateTableStatement();
```

Comment:
Do we add a UT to cover the branch of `!containsInTableContainedRule(tableName, materials)` ?

Reply:
Added.